monkey/infection_monkey/exploit/powershell.py

```diff
@@ -41,49 +47,79 @@ def _set_sensitive_packages_log_level_to_error():
             logging.getLogger(package.__name__).setLevel(logging.ERROR)
 
     def _exploit_host(self):
-        self.client = self._authenticate_via_brute_force()
+        is_https = self._is_client_using_https()
+        credentials = CredentialGenerator(
+            self.host.ip_addr,
+            self._config.exploit_user_list,
+            self._config.exploit_password_list,
+            is_windows_os(),
+        ).get_credentials(is_https=is_https)
```

Comment:
This doesn't need to be a class. This is effectively just a complicated function invocation. The class doesn't actually store any state.

Reply:
Agreed.